Source/WebCore/style/MatchedDeclarationsCache.cpp

```diff
@@ -133,23 +133,31 @@ unsigned MatchedDeclarationsCache::computeHash(const MatchResult& matchResult, c
     return WTF::computeHash(matchResult, &inheritedCustomProperties);
 }
 
-const MatchedDeclarationsCache::Entry* MatchedDeclarationsCache::find(unsigned hash, const MatchResult& matchResult, const StyleCustomPropertyData& inheritedCustomProperties)
+std::optional<MatchedDeclarationsCache::Result> MatchedDeclarationsCache::find(unsigned hash, const MatchResult& matchResult, const StyleCustomPropertyData& inheritedCustomProperties, const RenderStyle& parentStyle)
 {
     if (!hash)
-        return nullptr;
+        return std::nullopt;
 
     auto it = m_entries.find(hash);
     if (it == m_entries.end())
-        return nullptr;
+        return std::nullopt;
 
-    auto& entry = it->value;
-    if (!matchResult.cacheablePropertiesEqual(*entry.matchResult))
-        return nullptr;
+    Entry* result = nullptr;
```

Comment:
It would be nicer if the type of a variable called `result` was `Result` or `std::optional<Result>`. I think you can also simplify the flow a bit by using the latter since you can just return it at the end.